qirlib/src/metadata.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use llvm_sys::prelude::LLVMValueRef;
+
+use crate::llvm_wrapper::{LLVMRustExtractMDConstant, LLVMRustIsAMDConstant};
+
+pub unsafe fn extract_constant(value: LLVMValueRef) -> Option<LLVMValueRef> {
+    let constant_value = unsafe { LLVMRustExtractMDConstant(value) };
+    if constant_value.is_null() {
+        None
+    } else {
+        Some(constant_value)
+    }
+}
+
+pub unsafe fn is_constant(value: LLVMValueRef) -> bool {
+    let result = unsafe { LLVMRustIsAMDConstant(value) };
+    !result.is_null()
+}
```

Comment:
I think this can be removed? You can change the use site from:

```rust
        if !qirlib::metadata::is_constant(valueref) {
            Err(PyValueError::new_err("Value is not constant."))
        } else if let Some(constant) = qirlib::metadata::extract_constant(valueref) {
            // ...
        } else {
            // ...
        }
```

to

```rust
        if let Some(constant) = qirlib::metadata::extract_constant(valueref) {
            // ...
        } else {
            Err(PyValueError::new_err("Value is not constant."))
        }
```